### ISSUE TITLE
Question: BatchNorm2d you use vs. FrozenBatchNorm2d from Detectron2

### ISSUE THREAD

**Issue by PhilipMay** [lukemelas/EfficientNet-PyTorch#166]
Hi,
you are using `BatchNorm2d` in your implementation. [Detectron2](https://github.com/facebookresearch/detectron2) uses `FrozenBatchNorm2d` in its renet implementation. See here: https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/backbone/resnet.py

The reason they use `FrozenBatchNorm2d` seems to be this:
> The reason why we use FrozenBatchNorm2d instead of BatchNorm2d is that the sizes of the batches are very small, which makes the batch statistics very poor and degrades performance.
> Plus, when using multiple GPUs, the batch statistics are not accumulated from multiple devices, so that only a single GPU compute the statistics. 
> When we have sync BatchNorm in PyTorch, we could start looking into having BatchNorm instead of a frozen version of it. [from here](https://github.com/facebookresearch/maskrcnn-benchmark/issues/267#issuecomment-446571344)

How comes this difference here?

Thanks,
Philip

**Comment by lukemelas** [comment]
We use regular `BatchNorm2d` to match the official TensorFlow implementation. I agree that if you wanted to train from scratch, `FrozenBatchNorm2d` or a synchronized batchnorm would probably be better.
